src/main/java/nextstep/app/domain/CustomUserDetailsService.java

```diff
@@ -0,0 +1,20 @@
+package nextstep.app.domain;
+
+import nextstep.security.core.uesrdetails.UserDetails;
+import nextstep.security.core.uesrdetails.UserDetailsService;
+
+public class CustomUserDetailsService implements UserDetailsService {
+    private final MemberRepository memberRepository;
+
+    public CustomUserDetailsService(final MemberRepository memberRepository) {
+        this.memberRepository = memberRepository;
+    }
+
+    @Override
+    public UserDetails loadUserByUsername(final String username) {
+        Member member = memberRepository.findByEmail(username)
+                .orElseThrow(() -> new IllegalArgumentException("해당하는 사용자를 찾을 수 없습니다."));
+
+        return new CustomMember(member);
```

Comment:
```suggestion
        return memberRepository.findByEmail(username)
                .map(CustomMember::new)
                .orElseThrow(() -> new IllegalArgumentException("해당하는 사용자를 찾을 수 없습니다."));
```

소소한 가독성 상승을 해볼 수 있겠네요~